weed/iamapi/iamapi_handlers.go

```diff
@@ -1,31 +1,45 @@
 package iamapi
 
 import (
-	"fmt"
+	"net/http"
+
 	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/seaweedfs/seaweedfs/weed/glog"
 	"github.com/seaweedfs/seaweedfs/weed/s3api/s3err"
-	"net/http"
 )
 
-func writeIamErrorResponse(w http.ResponseWriter, r *http.Request, err error, object string, value string, msg error) {
-	errCode := err.Error()
```

Comment:
`writeIamErrorResponse` should not be responsible for creating error messages. It should only determine the HTTP Status code depending on the error code, and forward the message (or not) to the user.

The logic for determining the error code and message has been moved down to the respective IamApiServer functions.